src/generate-types.ts

```diff
@@ -0,0 +1,92 @@
+import type { CodegenPlugin } from '@graphql-codegen/plugin-helpers'
+
+module.exports = <CodegenPlugin> {
+  /**
+   * Generates TypeScript types based on the provided GraphQL schema.
+   * @param schema - The GraphQL schema object to generate types from.
+   * @param _documents - Unused parameter for additional documents.
+   * @param _config - Unused parameter for configuration options.
+   * @returns A string containing the generated TypeScript types.
+   */
+  plugin: function (schema, _documents, _config) {
+    if (!schema.getQueryType()) {
+      throw new Error('No query type found in schema')
+    }
+
+    /**
+     * Capitalizes the first letter of the given string.
+     * @param str - The string to capitalize.
+     * @returns The input string with the first letter converted to uppercase.
+     */
+    function capitalize (str: string) {
+      return str.charAt(0).toUpperCase() + str.slice(1)
+    }
```

Comment:
is this the best place for this `capitalize` util?

Reply:
I think so. I've given a thought or two, but since its the only place that we should use it, and it's totally coupled to just this functionality, I'd think it even makes more sense to leave it here rather than making a whole directory and file just for this function